tests/test_find_iocs.py

```diff
@@ -19,22 +23,26 @@ def test_find_iocs(text: str, results: Dict, args: Dict) -> None:
     ), f"Key values don't match '{ioc_finder_keys}' vs '{results_keys}'"
     for key in ioc_finder_keys:
         # Compare lists
-        if type(iocs[key]) == list:
-            _compare_lists(iocs[key], results[key])
+        if isinstance(iocs[key], list):
+            _compare_lists(iocs[key], results[key])  # type: ignore
         # Compare sub dictionaries like for attack patterns
-        elif type(iocs[key]) == dict:
-            for subkey in iocs[key].keys():
-                assert subkey in results[key], f"Expected key '{key}' not found in {results[key]}"
-                _compare_lists(iocs[key][subkey], results[key][subkey])
+        elif isinstance(iocs[key], dict):
+            _compare_dicts(iocs[key], results[key])  # type: ignore
```

Comment:
I slightly refactored the type checks and split the dict comparison into a separate function.